src/main/java/org/symphonyoss/symphony/messageml/elements/UIAction.java

```diff
@@ -0,0 +1,167 @@
+package org.symphonyoss.symphony.messageml.elements;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.symphonyoss.symphony.messageml.MessageMLContext;
+import org.symphonyoss.symphony.messageml.MessageMLParser;
+import org.symphonyoss.symphony.messageml.exceptions.InvalidInputException;
+import org.symphonyoss.symphony.messageml.util.XmlPrintStream;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class specify the Symphony Component UIAction which is represented by the tag name "ui-action".
+ * The messageML representation of this element can contain the following fields:
+ * <ul>
+ *   <li>trigger  -> default 'click', trigger of the action. For now only 'click' is supported</li>
+ *   <li>action (required) -> action that can be executed on trigger. For now only 'open-im' is supported</li>
+ * </ul>
+ * The open-im action can support the following attributes:
+ * <ul>
+ *   <li>user-ids (exclusive with stream-id)-> list of the users we want the action to be applied on</li>
+ *   <li>stream-id (exclusive with user-ids) -> stream we want the action to be applied on</li>
+ *   <li>side-by-side -> default true, open the result chat in a new module and keeps the parent one open side by side.
+ *                     If false, the parent one will be replaced.</li>
+ * <ul/>
+ * For now only Buttons and other UIActions will be allowed as child of a ui-action.
+ */
+
+public class UIAction extends Element {
+
+  public static final String MESSAGEML_TAG = "ui-action";
+
+  private static final String TRIGGER_ATTR = "trigger";
+  private static final String ACTION_ATTR = "action";
+  private static final String USER_IDS_ATTR = "user-ids";
+  private static final String STREAM_ID_ATTR = "stream-id";
+  private static final String SIDE_BY_SIDE_ATTR = "side-by-side";
+
+  private static final String DEFAULT_TRIGGER = "click";
+  private static final String ALLOWED_ACTION = "open-im";
+  private static final int MAX_USER_IDS = 15;
+
+  private static final String PRESENTATIONML_TAG = "div";
+  private static final String PRESENTATIONML_ACTION_ATTR = "data-action";
+  private static final String PRESENTATIONML_TRIGGER_ATTR = "data-trigger";
+  private static final String PRESENTATIONML_USER_IDS_ATTR = "data-user-ids";
+  private static final String PRESENTATIONML_STREAM_ID_ATTR = "data-stream-id";
+  private static final String PRESENTATIONML_SIDE_BY_SIDE_ATTR = "data-side-by-side";
+
+
+  private static final ObjectMapper MAPPER = new ObjectMapper();
```

Comment:
each element usually have its own mapper? they don't share a single one?

Reply:
I put the mapper in the Element object so it can be shared between all elements. https://github.com/symphonyoss/messageml-utils/pull/262/commits/8fed1fb980d6bd286508fce633da38572d7a1595